core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -154,8 +154,7 @@ public <T extends RESTResponse> T handleRequest(Route route, Map<String, String>
                                                   Object body, Class<T> responseType) {
     switch (route) {
       case CONFIG:
-        // TODO: use the correct response object
-        return castResponse(responseType, ImmutableMap.of());
+        return castResponse(responseType, CatalogHandlers.fetchServerConfig());
```

Comment:
I commented above that it probably isn't very helpful to include `fetchServerConfig()` in `CatalogHandlers`. I would recommend creating the response directly here. You could also include a default config map for testing purposes if you wanted, although I think you're testing it different way so you may not need it.

Reply:
Sure I can inline it here. That makes sense.

And since the server configuration being applied depends on what configuration is passed in, I chose to forgo it for now since it's tested elsewhere.